## ECUACIONES DE ESTADO
##### EJEMPLO 1: Compuestos puros


---


Este notebook corresponde a un ejemplo en donde se aplica la resolución de las ecuaciones de estado para obtener diferentes variables relacionadas con la cantidad de materia y diversas propiedades termodinámicas de compuestos puros.


Es recomendable revisar el archivo teórico (EOS_1_compuesto_puro.ipynb) para entender el funcionamiento de este programa y las formulas que se utilizan para cada ecuación de estado para calcular propiedades aplicados a compuestos puros. Así como es necesario revisar el archivo "README.md" para entender con mayor detalle acerca en que casos es mejor tomar en cuenta ciertas ecuaciones de estado.


Si se requiere modificar las tablas de resultados finales para cada métodos o en el resumen de resultados generales, vea el archivo "README.md" para ver como modificar los parámetros de salida.


**NOTA IMPORTANTE**: Para ejecutar este programa es necesario tener las librerías numpy y tabulate.



---


### Resolución de ejercicio


Determine el volumen molar ($\overline{V}$), densidad ($\rho$) y el factor de compresibilidad ($Z$) para los siguientes fluidos a una presión de 1.5 MPa y 500 K de temperatura.


a) Agua($H_{2}O$)$\;\;\;\;\;\;\;\;\;$b) Tolueno($C_{6}H_{5}CH_{3}$)$\;\;\;\;\;\;\;\;$c)Nitrógeno($N_{2}$)


DATOS DEL EJERCICIO


|Compuesto|Masa molar ($\overline{M},\;\frac{g}{mol}$)|Presión crítica ($P_{C},\;$atm)|Temperatura crítica ($T_{C},\;$K)|Factor acéntrico ($\omega$)|
|----------|----------|-|-|-|
| $H_{2}O$ | 18.015 | 218.30743 | 647.3 | 0.344 |
| $C_{6}H_{5}CH_{3}$ | 92.141 | 40.4639 | 591.8 | 0.263 |
| $N_{2}$    | 28.013 | 33.4567 | 126.2 | 0.039 |  


Fuente: Reid, R. C., Prausnitz, J. M., & Poling, B. E. (1987). The properties of gases & liquids (4 th edition). McGraw-Hill.


- Temperatura (T) = 500 K
- Presión (P) = 1.5 MPa = 14.8038 atm


Resolviendo el inciso a correspondiente al compuesto puero agua ($H_{2}O$).


---


### _**INGRESO DE DATOS**_


A continuación se muestra el panel de ingreso de datos para obtener los resultados correspondientes para cada una de las ecuaciones de estado. Para propósitos de este programa puede seguir la indicación mencionada anteriormente o ingresar los datos de las otras sustancias. 


En caso de requerir la variable volumen del sistema de referencia para agregar otras ecuaciones retire el comentario al inicio del reglón correspondiente.Se recomienda no cambiar el valor R (constante de los gases ideales), de hacerlo es necesario buscar valores de los datos a ingresar en las unidades que sean consistentes con los del nuevo R. El nombre de la sustancia (variable `COMP`) se escribe entre las comillas simples(`'oxígeno'`,`'ACETONA'`).

In [20]:
import numpy as np
import tabulate as tb

R=0.082 #(atm*L)/(mol*K) - CONSTANTE DE LOS GASES IDEALES

COMP='AGUA'        #NOMBRE DEL COMPUESTO
T=500              #(K)            - TEMPERATURA DEL SISTEMA DE REFERENCIA
P=14.8038          #(atm)          - PRESIÓN DEL SISTEMA DE REFERENCIA
#V=5.2             #(L)            - VOLUMEN DEL SISTEMA DE REFERENCIA (DESACTIVADA)
MMOLAR_GAS=18.015  #(g/mol)        - MASA MOLAR DEL COMPUESTO
PC=218.30743       #(atm)          - PRESIÓN CRITICO PARA EL COMPUESTO
TC=647.3           #(K)            - TEMPERATURA CRITICO PARA EL COMPUESTO
w_fa=0.344         #(ADIMENSIONAL) - FACTOR ACENTRICO

print('*** DATOS INGRESADOS AL PROGRAMA ***')

*** DATOS INGRESADOS AL PROGRAMA ***


---


# Ecuación del gas ideal


En el caso de la ecuación de los gases ideales no hace distinción entre los compuestos por lo que las ecuaciones no toman en cuenta las interacciones moleculares dentro de la mezcla de gases. A partir de la ecuación del gas ideal:


\begin{equation}
    P\overline{V}=nRT
\end{equation}


Las variables que se presentan en las ecuaciones:  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del gas  
$R =$ Constante del gas ideal  
$n =$ Cantidad de moles del compuesto


Como primer paso es obtener el volumen molar, a partir de la suposición de que el número de moles $n$ sea 1, entonces el volumen molar queda como:


\begin{equation}
    \overline{V}=\frac{RT}{P}
\end{equation}


Como segundo paso es obtener la densidad del gas ($\rho$) y el factor de compresibilidad (Z)


\begin{equation}
    \rho=\frac{P\overline{M}}{RT}
\end{equation}


\begin{equation}
    Z=\frac{P\overline{V}}{RT}
\end{equation}


Las variables que se presentan en las ecuaciones:  
$\rho=$ Densidad de la sustancia  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del compuesto  
$\overline{M}=$ Masa molar del compuesto  
$R =$ Constante del gas ideal  
$Z =$ Factor de compresibilidad


Un detalle importante es que el factor de compresibilidad ($Z$) es de 1 debido a que en condiciones ideales no hay compresibilidad debido a que no se consideran las interacciones intermoleculares entre las moléculas.

In [21]:
print("*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DEL GAS IDEAL ***\n")
VM_EGI=(R*T)/P
DENS_EGI=(P*MMOLAR_GAS)/(R*T)
Z_EGI=(P*VM_EGI)/(R*T)

# - SALIDA DE RESULTADOS
print("RESULTADOS PARA LA ECUACIÓN DEL GAS IDEAL")
print("--------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB2_RES = ['VOLUMEN MOLAR DEL FLUIDO \n(V, L/mol)','DENSIDAD DEL FLUIDO \n(ρ, kg/m^3)','FACTOR DE COMPRESIBILIDAD \n(Z)']
R_VAR_EGI = [VM_EGI, DENS_EGI, Z_EGI]
FIL_R_EGI = []
for i in range(3):
    FIL_R_EGI.append([VAR_TAB2_RES[i],R_VAR_EGI[i]])
print(tb.tabulate(FIL_R_EGI, headers=ENC_RES, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))
print("\n------------------------------------------------------\n\n\n")


*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DEL GAS IDEAL ***

RESULTADOS PARA LA ECUACIÓN DEL GAS IDEAL
--------------------------------------------------------

╒════════════════════════════╤═════════════╕
│          VARIABLE          │  RESULTADO  │
╞════════════════════════════╪═════════════╡
│ VOLUMEN MOLAR DEL FLUIDO   │  2.769559   │
│         (V, L/mol)         │             │
├────────────────────────────┼─────────────┤
│    DENSIDAD DEL FLUIDO     │  6.504645   │
│        (ρ, kg/m^3)         │             │
├────────────────────────────┼─────────────┤
│ FACTOR DE COMPRESIBILIDAD  │  1.000000   │
│            (Z)             │             │
╘════════════════════════════╧═════════════╛

------------------------------------------------------





#### Para la ecucación del gas ideal se obtiene que en las condiciones del problema planteado la _**densidad es de 6.50 kg/m³ y un factor de compresibilidad de 1**_ . Sin embargo, como se mencionó anteriormente este cálculo no toma en cuenta el volumen molar que tenga la mezcla incluyendo las interacciones intemoleculares que tengan entre las especies químicas.

---


# Ecuación de estado de Van der Waals


A continuación se muestra los pasos para realizar los cálculos correspondientes para la ecuación de estado de Van der Waals para resolver el ejercicio. Los resultados de cada paso se muestran en los datos de salida de la celda del código.


1. Se obtienen los parámetros de la ecuación de estado $a$ y $b$ del compuesto


\begin{equation}
    a\;=\;\frac{27R^{2}T^{2}_{C}}{64P_{C}}\;\;\;\;\;\;\;\;b\;=\;\frac{RT_{C}}{8P_{C}}
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:   
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado  
$b =$ Parámetro b para la ecuación de estado  
$T_{C}=$ Temperatura crítica del compuesto  
$P_{C}=$ Presión crítica del compuesto


2. Una vez obtenido los parámetros $a$ y $b$ para el compuesto puro de la ecuación de estado se resuelve la ecuación polinomial en términos del volumen molar ($\overline{V}$) sustituyendo todos los datos tanto de presión y temperatura del sistema; constante de los gases ideales; y los parámetros $a$ y $b$.


\begin{equation}
    \overline{V^{3}}\;-\;\left(\;b\;+\;\frac{RT}{P}\;\right)\;\overline{V^{2}}\;+\;\left(\;\frac{a}{P}\;\right)\;\overline{V}\;-\;\left(\;\frac{ab}{P}\;\right)\;=\;0
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del gas  
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado del compuesto   
$b =$ Parámetro b para la ecuación de estado del compuesto


3. Una vez obtenido los coeficientes de ecuación polinomial solo es obtener la raíz real única del misma. (NOTA: Hay en ciertos casos en donde se puede obtener tres raíces reales solo se toma la mayor que corresponde el volumen molar del gas.)


4. Luego de haber obtenido el volumen molar ($\overline{V}$) se puede obtener la densidad de la compuesto ($\rho$) y también el factor de compresibilidad del compuesto en las condiciones propuestas en el principio a partir de las siguientes ecuaciones.


\begin{equation}
    \rho=\frac{P\overline{M}}{RT}
\end{equation}


\begin{equation}
    Z=\frac{P\overline{V}}{RT}
\end{equation}


Las variables que se presentan en las ecuaciones:  
$\rho=$ Densidad de la sustancia  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del compuesto  
$\overline{M}=$ Masa molar del compuesto  
$R =$ Constante del gas ideal  
$Z =$ Factor de compresibilidad

In [22]:
print("*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE VAN DER WAALS ***\n\n")

# - CALCULAR LAS CONSTANTE a Y b PARA COMPONENTE PURO
a_VW=((27*pow(R,2)*pow(TC,2))/(64*PC))
b_VW=((R*TC)/(8*PC))

print(f"1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Van der Waals \nPARA EL COMPUESTO {COMP.upper()}")
print("------------------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB1_RES = ['CONSTANTE a','CONSTANTE b']
R_VAR_EEVW = [a_VW, b_VW]
FIL_R_EEVW = []
for i in range(2):
    FIL_R_EEVW.append([VAR_TAB1_RES[i],R_VAR_EEVW[i]])
print(tb.tabulate(FIL_R_EEVW, headers=ENC_RES, tablefmt="fancy_grid", floatfmt="5.6f", stralign="center", numalign="center"))
print('\n\n')

# -  CALCULAR LOS COEFICIENTES DE LA ECUACIÓN DE ESTADO DE VAN DER WAALS EN SU FORMA POLINOMIAL 
print("2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO")
print("------------------------------------------------------------------------------\n")
coeff1_VW=1                       #TERMINO CÚBICO
coeff2_VW=-1*(b_VW+((R*T)/P))     #TERMINO CUADRÁTICO
coeff3_VW=(a_VW/P)                #TERMINO LINEAL
coeff4_VW=-1*((a_VW*b_VW)/P)      #TERMINO INDEPENDIENTE

POL_VW=[coeff1_VW,coeff2_VW,coeff3_VW,coeff4_VW]
print("--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:\n")
print("%5.1f V^3 + \t%5.10f V^2 + \t%5.10f V + \t%5.10f = 0\n\n"%(POL_VW[0],POL_VW[1],POL_VW[2],POL_VW[3]))

# - OBTENER LA RAÍZ REAL DE LA ECUACIÓN POLINOMIAL Y POR TANTO EL VOLUMEN MOLAR
EPOL_VW=np.roots(POL_VW)
print("--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:\n")
print(EPOL_VW)
print("\n\n--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:\n")
# - SELECCIÓN DE LA RAIZ REAL DE LA ECUACIÓN POLINOMIAL
RR_VW=sum(np.isreal(EPOL_VW))
if RR_VW == 1:
    VM_VW=EPOL_VW[np.isreal(EPOL_VW)].real[0]
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_VW))
    print("\n\n")
elif RR_VW == 3:
    VM_VW=max(EPOL_VW)
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_VW))
    print("\n\n")
else:
    print("*** ERROR NO HAY SOLUCION PARA ESTA EOS EN ESTAS CONDICIONES, NO HAY RAIZ REAL UNICA ***\n\n")

# - CALCULO DE LA DENSIDAD Y DEL FACTOR DE COMPRESIBILIDAD PARA LA SUSTANCIA
DENS_VW=MMOLAR_GAS/VM_VW
Z_VW=(P*VM_VW)/(R*T)

# - SALIDA DE RESULTADOS
print("4. RESULTADOS PARA LA ECUACIÓN DE ESTADO DE VAN DER WAALS")
print("---------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB2_RES = ['VOLUMEN MOLAR DEL FLUIDO \n(V, L/mol)','DENSIDAD DEL FLUIDO \n(ρ, kg/m^3)','FACTOR DE COMPRESIBILIDAD \n(Z)']
R_VAR_EEVW = [VM_VW, DENS_VW, Z_VW]
FIL_R_EEVW = []
for i in range(3):
    FIL_R_EEVW.append([VAR_TAB2_RES[i],R_VAR_EEVW[i]])
print(tb.tabulate(FIL_R_EEVW, headers=ENC_RES, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))
print("\n-------------------------------------------------------------------\n\n\n")


*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE VAN DER WAALS ***


1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Van der Waals 
PARA EL COMPUESTO AGUA
------------------------------------------------------------------------------

╒═════════════╤═════════════╕
│  VARIABLE   │  RESULTADO  │
╞═════════════╪═════════════╡
│ CONSTANTE a │  5.444452   │
├─────────────┼─────────────┤
│ CONSTANTE b │  0.030392   │
╘═════════════╧═════════════╛



2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO
------------------------------------------------------------------------------

--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:

  1.0 V^3 + 	-2.7999512844 V^2 + 	0.3677739658 V + 	-0.0111774295 = 0


--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:

[2.66344485 0.08974497 0.04676147]


--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:

VOLUMEN MOLAR DEL FLUIDO (L/mol) = 2.663445



4. RESULTADOS PARA L

#### Para la ecuación de estado de Van der Waals se obtiene que en las condiciones del problema planteado la _**densidad es de 6.76 kg/m³ y un factor de compresibilidad de 0.9616**_ .

---


# Ecuación de estado de Redlich-Kwong


A continuación se muestra los pasos para realizar los cálculos correspondientes para la ecuación de estado de Redlich-Kwong para resolver el ejercicio. Los resultados de cada paso se muestran en los datos de salida de la celda del código.


1. Se obtienen los parámetros de la ecuación de estado $a$ y $b$ del compuesto


\begin{equation}
    a\;=\;\frac{0.42748\;R^{2}T^{2.5}_{C}}{P_{C}}\;\;\;\;\;\;\;\;b\;=\;\frac{0.08664\;RT_{C}}{P_{C}}
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:   
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado del compuesto  
$b =$ Parámetro b para la ecuación de estado del compuesto  
$T_{C}=$ Temperatura crítica del compuesto  
$P_{C}=$ Presión crítica del compuesto


2. Una vez obtenido los parámetros de la ecuación de estado se resuelve la ecuación polinomial en términos del volumen molar ($\overline{V}$) sustituyendo todos los datos tanto de presión y temperatura del sistema; constante de los gases ideales; y los parámetros $a$ y $b$.


\begin{equation}
    \overline{V^{3}}\;-\;\left(\;\frac{RT}{P}\;\right)\;\overline{V^{2}}\;+\;\frac{1}{P}\;\left(\;\frac{a}{\sqrt{T}}\;-\;bRT\;-\;Pb^{2}\;\right)\;\overline{V}\;-\;\left(\;\frac{ab}{P\sqrt{T}}\;\right)\;=\;0
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del gas  
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado del compuesto  
$b =$ Parámetro b para la ecuación de estado del compuesto


3. Una vez obtenido los coeficientes de ecuación polinomial solo es obtener la raíz real única del misma. (NOTA: Hay en ciertos casos en donde se puede obtener tres raíces reales solo se toma la mayor que corresponde el volumen molar del gas.)


4. Luego de haber obtenido el volumen molar ($\overline{V}$) se puede obtener la densidad de la compuesto ($\rho$) y también el factor de compresibilidad del compuesto en las condiciones propuestas en el principio a partir de las siguientes ecuaciones.


\begin{equation}
    \rho=\frac{P\overline{M}}{RT}
\end{equation}


\begin{equation}
    Z=\frac{P\overline{V}}{RT}
\end{equation}


Las variables que se presentan en las ecuaciones:  
$\rho=$ Densidad de la sustancia  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del compuesto  
$\overline{M}=$ Masa molar del compuesto  
$R =$ Constante del gas ideal  
$Z =$ Factor de compresibilidad  

In [23]:
print("*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE REDLICH-KWONG ***\n")

# - CALCULAR LAS CONSTANTE a Y b PARA COMPONENTE DE LA MEZCLA
a_RK=((0.42748*pow(R,2)*pow(TC,2.5))/PC)
b_RK=((0.08664*R*TC)/PC)

print(f"1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Redlich-Kwong \nPARA EL COMPUESTO {COMP.upper()}")
print("------------------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB1_RES = ['CONSTANTE a','CONSTANTE b']
R_VAR_EERK = [a_RK, b_RK]
FIL_R_EERK = []
for i in range(2):
    FIL_R_EERK.append([VAR_TAB1_RES[i],R_VAR_EERK[i]])
print(tb.tabulate(FIL_R_EERK, headers=ENC_RES, tablefmt="fancy_grid", floatfmt="5.6f", stralign="center", numalign="center"))
print('\n\n')

# - CALCULAR LOS COEFICIENTES DE LA ECUACIÓN DE ESTADO DE REDLICH-KWONG EN SU FORMA POLINOMIAL
print("2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO")
print("------------------------------------------------------------------------------\n")
coeff1_RK=1                                               #TERMINO CÚBICO
coeff2_RK=-1*((R*T)/P)                                    #TERMINO CUADRÁTICO
coeff3_RK=(1/P)*((a_RK/T**0.5)-(b_RK*R*T)-(P*(b_RK**2)))  #TERMINO LINEAL
coeff4_RK=-1*((a_RK*b_RK)/(P*(T**0.5)))                   #TERMINO INDEPENDIENTE

POL_RK=[coeff1_RK,coeff2_RK,coeff3_RK,coeff4_RK]
print("--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:\n")
print("%5.1f V^3 + \t%5.10f V^2 + \t%5.10f V + \t%5.10f = 0\n\n"%(POL_RK[0],POL_RK[1],POL_RK[2],POL_RK[3]))

# - OBTENER LA RAÍZ REAL DE LA ECUACIÓN POLINOMIAL Y POR TANTO EL VOLUMEN MOLAR
EPOL_RK=np.roots(POL_RK)
print("--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:\n")
print(EPOL_RK)
print("\n\n--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:\n")
# - SELECCIÓN DE LA RAIZ REAL DE LA ECUACIÓN POLINOMIAL
RR_RK=sum(np.isreal(EPOL_RK))
if RR_RK == 1:
    VM_RK=EPOL_RK[np.isreal(EPOL_RK)].real[0]
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_RK))
    print("\n\n")
elif RR_RK == 3:
    VM_RK=max(EPOL_RK)
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_RK))
    print("\n\n")
else:
    print("*** ERROR NO HAY SOLUCION PARA ESTA EOS EN ESTAS CONDICIONES, NO HAY RAIZ REAL UNICA ***\n\n")

# - CALCULO DE LA DENSIDAD Y DEL FACTOR DE COMPRESIBILIDAD PARA LA SUSTANCIA
DENS_RK=MMOLAR_GAS/VM_RK
Z_RK=(P*VM_RK)/(R*T)

# - RESULTADOS DE SALIDA
print("4. RESULTADOS PARA LA ECUACIÓN DE ESTADO DE REDLICH-KWONG")
print("---------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB2_RES = ['VOLUMEN MOLAR DEL FLUIDO \n(V, L/mol)','DENSIDAD DEL FLUIDO \n(ρ, kg/m^3)','FACTOR DE COMPRESIBILIDAD \n(Z)']
R_VAR_EERK = [VM_RK, DENS_RK, Z_RK]
FIL_R_EERK = []
for i in range(3):
    FIL_R_EERK.append([VAR_TAB2_RES[i],R_VAR_EERK[i]])
print(tb.tabulate(FIL_R_EERK, headers=ENC_RES, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))
print("\n-------------------------------------------------------------------\n\n\n")


*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE REDLICH-KWONG ***

1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Redlich-Kwong 
PARA EL COMPUESTO AGUA
------------------------------------------------------------------------------

╒═════════════╤═════════════╕
│  VARIABLE   │  RESULTADO  │
╞═════════════╪═════════════╡
│ CONSTANTE a │ 140.358593  │
├─────────────┼─────────────┤
│ CONSTANTE b │  0.021065   │
╘═════════════╧═════════════╛



2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO
------------------------------------------------------------------------------

--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:

  1.0 V^3 + 	-2.7695591672 V^2 + 	0.3652290058 V + 	-0.0089320302 = 0


--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:

[2.63208829 0.10521894 0.03225194]


--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:

VOLUMEN MOLAR DEL FLUIDO (L/mol) = 2.632088



4. RESULTADOS PARA LA

#### Para la ecuación de estado de Redlich-Kwong se obtiene que en las condiciones del problema planteado la _**densidad es de 6.84 kg/m³ y un factor de compresibilidad de 0.9503**_ .

---


# Ecuación de estado de Soave


A continuación se muestra los pasos para realizar los cálculos correspondientes para la ecuación de estado de Soave para resolver el ejercicio. Los resultados de cada paso se muestran en los datos de salida de la celda del código.


1. Se obtienen los parámetros de la ecuación de estado $a$, $b$ y $\alpha$ del compuesto


\begin{equation}
    a\;=\;\frac{0.42747\;R^{2}T^{2}_{C}}{P_{C}}\;\;\;\;\;\;\;\;b\;=\;\frac{0.08664\;RT_{C}}{P_{C}}
\end{equation}


\begin{equation}
    \alpha\;=\;\left[1\;+\;\left(\left(0.48508\;+\;1.55171\;\omega\;-\;0.15613\;\omega^{2}\right)\left(1\;-\;T_{r}^{0.5}\right)\right)\right]^{2}
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:   
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado  
$b =$ Parámetro b para la ecuación de estado  
$\alpha =$ Parámetro $\alpha$ para la ecuación de estado  
$T_{C}=$ Temperatura crítica del compuesto  
$P_{C}=$ Presión crítica del compuesto  
$\omega=$ Factor acéntrico del compuesto  
$T_{r}\;=\;\dfrac{T}{T_{C}}\;=$ Temperatura reducida es la división entre la temperatura del sistema de referencia y la temperatura crítica del compuesto  


2. Una vez obtenido los parámetros de la ecuación de estado se resuelve la ecuación polinomial en términos del volumen molar ($\overline{V}$) sustituyendo todos los datos tanto de presión y temperatura del sistema; constante de los gases ideales; y los parámetros $a$ y $b$.


\begin{equation}
    \overline{V^{3}}\;-\;\left(\;\frac{RT}{P}\;\right)\;\overline{V^{2}}\;+\;\frac{1}{P}\;\left(\;a\;\alpha\;-\;bRT\;-\;Pb^{2}\;\right)\;\overline{V}\;-\;\left(\;\frac{a\alpha b}{P}\;\right)\;=\;0
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del gas  
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado  
$b =$ Parámetro b para la ecuación de estado  
$\alpha =$ Parámetro $\alpha$ para la ecuación de estado  


3. Una vez obtenido los coeficientes de ecuación polinomial solo es obtener la raíz real única del misma. (NOTA: Hay en ciertos casos en donde se puede obtener tres raíces reales solo se toma la mayor que corresponde el volumen molar del gas.)


4. Luego de haber obtenido el volumen molar ($\overline{V}$) se puede obtener la densidad de la compuesto ($\rho$) y también el factor de compresibilidad del compuesto en las condiciones propuestas en el principio a partir de las siguientes ecuaciones.


\begin{equation}
    \rho=\frac{P\overline{M}}{RT}
\end{equation}


\begin{equation}
    Z=\frac{P\overline{V}}{RT}
\end{equation}


Las variables que se presentan en las ecuaciones:  
$\rho=$ Densidad de la sustancia  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del compuesto  
$\overline{M}=$ Masa molar del compuesto  
$R =$ Constante del gas ideal  
$Z =$ Factor de compresibilidad  

In [24]:
print("*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE SOAVE ***\n")

#* CALCULAR LAS CONSTANTE a Y b PARA COMPONENTE DE LA MEZCLA
TR_S=T/TC
a_S=((0.42747*pow(R,2)*pow(TC,2))/PC)
b_S=((0.08664*R*TC)/PC)
alpha_S=(1+(0.48508+(1.55171*(w_fa))-(0.15613*(w_fa**2)))*(1-(TR_S**0.5)))**2

print(f"1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Soave \nPARA EL COMPUESTO {COMP.upper()}")
print("------------------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB1_RES = ['CONSTANTE a','CONSTANTE b','CONSTANTE alpha', 'TEMPERATURA REDUCIDA']
R_VAR_EES = [a_S, b_S, alpha_S, TR_S]
FIL_R_EES = []
for i in range(4):
    FIL_R_EES.append([VAR_TAB1_RES[i],R_VAR_EES[i]])
print(tb.tabulate(FIL_R_EES, headers=ENC_RES, tablefmt="fancy_grid", floatfmt="5.6f", stralign="center", numalign="center"))
print('\n\n')

# - CALCULAR LOS COEFICIENTES DE LA ECUACIÓN DE ESTADO DE SOAVE EN SU FORMA POLINOMIAL
print("2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO")
print("------------------------------------------------------------------------------\n")
coeff1_S=1                                              #TERMINO CÚBICO
coeff2_S=-1*((R*T)/P)                                   #TERMINO CUADRÁTICO
coeff3_S=(1/P)*((a_S*alpha_S)-(b_S*R*T)-(P*(b_S**2)))   #TERMINO LINEAL
coeff4_S=-1*((a_S*alpha_S*b_S)/(P))                     #TERMINO INDEPENDIENTE

POL_S=[coeff1_S,coeff2_S,coeff3_S,coeff4_S]
print("--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:\n")
print("%5.1f V^3 + \t%5.10f V^2 + \t%5.10f V + \t%5.10f = 0\n\n"%(POL_S[0],POL_S[1],POL_S[2],POL_S[3]))

# - OBTENER LA RAÍZ REAL DE LA ECUACIÓN POLINOMIAL Y POR TANTO EL VOLUMEN MOLAR
EPOL_S=np.roots(POL_S)
print("--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:\n")
print(EPOL_S)
print("\n\n--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:\n")
# - SELECCIÓN DE LA RAIZ REAL DE LA ECUACIÓN POLINOMIAL
RR_S=sum(np.isreal(EPOL_S))
if RR_S == 1:
    VM_S=EPOL_S[np.isreal(EPOL_S)].real[0]
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_S))
    print("\n\n")
elif RR_S == 3:
    VM_S=max(EPOL_S)
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_S))
    print("\n\n")
else:
    print("*** ERROR NO HAY SOLUCION PARA ESTA EOS EN ESTAS CONDICIONES, NO HAY RAIZ REAL UNICA ***\n\n")

# - CALCULO DE LA DENSIDAD Y DEL FACTOR DE COMPRESIBILIDAD PARA LA SUSTANCIA
DENS_S=MMOLAR_GAS/VM_S
Z_S=(P*VM_S)/(R*T)

# - RESULTADOS DE SALIDA
print("4. RESULTADOS PARA LA ECUACIÓN DE ESTADO DE SOAVE")
print("---------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB2_RES = ['VOLUMEN MOLAR DEL FLUIDO \n(V, L/mol)','DENSIDAD DEL FLUIDO \n(ρ, kg/m^3)','FACTOR DE COMPRESIBILIDAD \n(Z)']
R_VAR_EES = [VM_S, DENS_S, Z_S]
FIL_R_EES = []
for i in range(3):
    FIL_R_EES.append([VAR_TAB2_RES[i],R_VAR_EES[i]])
print(tb.tabulate(FIL_R_EES, headers=ENC_RES, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))
print("\n-------------------------------------------------------------------\n\n\n")


*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE SOAVE ***

1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Soave 
PARA EL COMPUESTO AGUA
------------------------------------------------------------------------------

╒══════════════════════╤═════════════╕
│       VARIABLE       │  RESULTADO  │
╞══════════════════════╪═════════════╡
│     CONSTANTE a      │  5.516658   │
├──────────────────────┼─────────────┤
│     CONSTANTE b      │  0.021065   │
├──────────────────────┼─────────────┤
│   CONSTANTE alpha    │  1.257005   │
├──────────────────────┼─────────────┤
│ TEMPERATURA REDUCIDA │  0.772439   │
╘══════════════════════╧═════════════╛



2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO
------------------------------------------------------------------------------

--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:

  1.0 V^3 + 	-2.7695591672 V^2 + 	0.4096390862 V + 	-0.0098675456 = 0


--> 3. LAS 

#### Para la ecuación de estado de Soave se obtiene que en las condiciones del problema planteado la _**densidad es de 6.89 kg/m³ y un factor de compresibilidad de 0.9439**_ .

---


# Ecuación de estado de Peng-Robinson


A continuación se muestra los pasos para realizar los cálculos correspondientes para la ecuación de estado de Peng-Robinson para resolver el ejercicio. Los resultados de cada paso se muestran en los datos de salida de la celda del código.


1. Se obtienen los parámetros de la ecuación de estado $a$, $b$ y $\alpha$ del compuesto


\begin{equation}
    a\;=\;\frac{0.45724\;R^{2}T^{2}_{C}}{P_{C}}\;\;\;\;\;\;\;\;b\;=\;\frac{0.07780\;RT_{C}}{P_{C}}
\end{equation}


\begin{equation}
    \alpha\;=\;\left[1\;+\;\left(\left(0.37464\;+\;1.54226\;\omega\;-\;0.26992\;\omega^{2}\right)\left(1\;-\;T_{r}^{0.5}\right)\right)\right]^{2}
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:   
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado  
$b =$ Parámetro b para la ecuación de estado  
$\alpha =$ Parámetro $\alpha$ para la ecuación de estado  
$T_{C}=$ Temperatura crítica del compuesto  
$P_{C}=$ Presión crítica del compuesto  
$\omega=$ Factor acéntrico del compuesto  
$T_{r}\;=\;\dfrac{T}{T_{C}}\;=$ Temperatura reducida es la división entre la temperatura del sistema de referencia y la temperatura crítica del compuesto


2. Una vez obtenido los parámetros de la ecuación de estado se resuelve la ecuación polinomial en términos del volumen molar ($\overline{V}$) sustituyendo todos los datos tanto de presión y temperatura del sistema; constante de los gases ideales; y los parámetros $a$ y $b$.


\begin{equation}
    \overline{V^{3}}\;+\;\left(\;b\;-\;\frac{RT}{P}\;\right)\;\overline{V^{2}}\;+\;\left(\;\frac{a\alpha}{P}\;-\;\frac{2bRT}{P}\;-\;3b^{2}\right)\;\overline{V}\;+\;\left(b^{3}\;+\;\frac{RTb^{2}}{P}\;-\;\frac{a\alpha b}{P}\right)\;=\;0
\end{equation}


Las variables que se presentan en las ecuaciones son las siguientes:  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del gas  
$R =$ Constante del gas ideal  
$a =$ Parámetro a para la ecuación de estado  
$b =$ Parámetro b para la ecuación de estado  
$\alpha =$ Parámetro $\alpha$ para la ecuación de estado  


3. Una vez obtenido los coeficientes de ecuación polinomial solo es obtener la raíz real única del misma. (NOTA: Hay en ciertos casos en donde se puede obtener tres raíces reales solo se toma la mayor que corresponde el volumen molar del gas.)


4. Luego de haber obtenido el volumen molar ($\overline{V}$) se puede obtener la densidad de la compuesto ($\rho$) y también el factor de compresibilidad del compuesto en las condiciones propuestas en el principio a partir de las siguientes ecuaciones.


\begin{equation}
    \rho=\frac{P\overline{M}}{RT}
\end{equation}


\begin{equation}
    Z=\frac{P\overline{V}}{RT}
\end{equation}


Las variables que se presentan en las ecuaciones:  
$\rho=$ Densidad de la sustancia  
$P=$ Presión del sistema de referencia  
$T=$ Temperatura del sistema de referencia  
$\overline{V}=$ Volumen molar del compuesto  
$\overline{M}=$ Masa molar del compuesto  
$R =$ Constante del gas ideal  
$Z =$ Factor de compresibilidad

In [25]:
print("*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE PENG-ROBINSON ***\n")

# - CALCULAR LAS CONSTANTE a Y b PARA COMPONENTE DE LA MEZCLA
TR_PR=T/TC
a_PR=((0.45724*pow(R,2)*pow(TC,2))/PC)
b_PR=((0.07780*R*TC)/PC)
alpha_PR=(1+(0.37464+(1.54226*(w_fa))-(0.26992*(w_fa**2)))*(1-(TR_PR**0.5)))**2

print(f"1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Soave \nPARA EL COMPUESTO {COMP.upper()}")
print("------------------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB1_RES = ['CONSTANTE a','CONSTANTE b','CONSTANTE alpha', 'TEMPERATURA REDUCIDA']
R_VAR_EEPR = [a_PR, b_PR, alpha_PR, TR_PR]
FIL_R_EEPR = []
for i in range(4):
    FIL_R_EEPR.append([VAR_TAB1_RES[i],R_VAR_EEPR[i]])
print(tb.tabulate(FIL_R_EEPR, headers=ENC_RES, tablefmt="fancy_grid", floatfmt="5.6f", stralign="center", numalign="center"))
print('\n\n')

# - CALCULAR LOS COEFICIENTES DE LA ECUACIÓN DE ESTADO DE PENG-ROBINSON EN SU FORMA POLINOMIAL
print("2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO")
print("------------------------------------------------------------------------------\n")
coeff1_PR=1                                                          #TERMINO CÚBICO
coeff2_PR=((b_PR)-((R*T)/P))                                         #TERMINO CUADRÁTICO
coeff3_PR=(((a_PR*alpha_PR)/P)-((2*b_PR*R*T)/P)-(3*(b_PR**2)))       #TERMINO LINEAL
coeff4_PR=((b_PR**3)+((R*T*(b_PR**2))/P)-((a_PR*alpha_PR*b_PR)/P))   #TERMINO INDEPENDIENTE

POL_PR=[coeff1_PR,coeff2_PR,coeff3_PR,coeff4_PR]
print("--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:\n")
print("%5.5f V^3 + \t%5.10f V^2 + \t%5.10f V + \t%5.10f = 0\n\n"%(POL_PR[0],POL_PR[1],POL_PR[2],POL_PR[3]))

# - OBTENER LA RAÍZ REAL DE LA ECUACIÓN POLINOMIAL Y POR TANTO EL VOLUMEN MOLAR
EPOL_PR=np.roots(POL_PR)
print("--> 3. LAS RAICES OBTENIDAS DE LA ECUACIÓN DE POLINOMIAL SON:\n")
print(EPOL_PR)
print("\n\n--> DE LO OBTENIDO DE LA ECUACIÓN DE ESTADO:\n")
# - SELECCIÓN DE LA RAIZ REAL DE LA ECUACIÓN POLINOMIAL
RR_PR=sum(np.isreal(EPOL_PR))
if RR_PR == 1:
    VM_PR=EPOL_PR[np.isreal(EPOL_PR)].real[0]
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_PR))
    print("\n\n")
elif RR_PR == 3:
    VM_PR=max(EPOL_PR)
    print("VOLUMEN MOLAR DEL FLUIDO (L/mol) = %5.6f"%(VM_S))
    print("\n\n")
else:
    print("*** ERROR NO HAY SOLUCION PARA ESTA EOS EN ESTAS CONDICIONES, NO HAY RAIZ REAL UNICA ***\n\n")

# - OBTENIENDO LOS MOLES Y MASA DEL GAS
DENS_PR=MMOLAR_GAS/VM_PR
Z_PR=(P*VM_PR)/(R*T)

# - RESULTADOS DE SALIDA
print("4. RESULTADOS PARA LA ECUACIÓN DE ESTADO DE PENG-ROBINSON")
print("-------------------------------------------------------------------------\n")
ENC_RES = ["VARIABLE", "RESULTADO"]
VAR_TAB2_RES = ['VOLUMEN MOLAR DEL FLUIDO (L/mol)','DENSIDAD DEL FLUIDO (kg/m^3)','FACTOR DE COMPRESIBILIDAD (Z)']
R_VAR_EEPR = [VM_PR, DENS_PR, Z_PR]
FIL_R_EEPR = []
for i in range(3):
    FIL_R_EEPR.append([VAR_TAB2_RES[i],R_VAR_EEPR[i]])
print(tb.tabulate(FIL_R_EEPR, headers=ENC_RES, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))
print("\n--------------------------------------------------------------------------\n\n\n")


*** CALCULANDO LAS VARIABLES DEL FLUIDO POR LA ECUACIÓN DE ESTADO DE PENG-ROBINSON ***

1. OBTENCIÓN DE LAS CONSTANTES a y b PARA LA ECUACIÓN DE ESTADO DE Soave 
PARA EL COMPUESTO AGUA
------------------------------------------------------------------------------

╒══════════════════════╤═════════════╕
│       VARIABLE       │  RESULTADO  │
╞══════════════════════╪═════════════╡
│     CONSTANTE a      │  5.900851   │
├──────────────────────┼─────────────┤
│     CONSTANTE b      │  0.018916   │
├──────────────────────┼─────────────┤
│   CONSTANTE alpha    │  1.222709   │
├──────────────────────┼─────────────┤
│ TEMPERATURA REDUCIDA │  0.772439   │
╘══════════════════════╧═════════════╛



2. CALCULO DE LOS COEFICIENTES DE LA ECUACIÓN POLINOMIAL Y DE LA RAIZ DEL MISMO
------------------------------------------------------------------------------

--> LA ECUACIÓN POLINOMIAL CALCULADA EN TÉRMINOS DE VOLUMEN MOLAR:

1.00000 V^3 + 	-2.7506431135 V^2 + 	0.3815247255 V + 	-0.0082214748 = 0


-

#### Para la ecuación de estado de Peng-Robinson se obtiene que en las condiciones del problema planteado la _**densidad es de 6.91 kg/m³ y un factor de compresibilidad de 0.9407**_ .

---


# TABLA COMPARATIVA DE RESULTADOS ENTRE LOS DIFERENTES ECUACIONES DE ESTADO

In [26]:
#RESUMEN DE RESULTADOS DE LO OBTENIDO EN ESTE PROGRAMA
print("RESUMEN DE RESULTADOS DE VOLUMEN MOLAR, DENSIDAD Y FACTOR DE COMPRESIBILIDAD DEL \nGAS PURO POR CADA ECUACION DE ESTADO")
print("---------------------------------------------------------------------------------\n")

ENC_RVD = ["ECUACIÓN DE\nESTADO", "VOLUMEN MOLAR\n(L/mol)", "DENSIDAD\n(kg/m^3)","FACTOR DE\nCOMPRESIBILIDAD"]
EOS=['Gas ideal','Van der Waals', 'Redlich-Kwong', 'Soave', 'Peng-Robinson']
VM_SIST=[VM_EGI, VM_VW, VM_RK, VM_S, VM_PR]
DENS_SIST=[DENS_EGI, DENS_VW, DENS_RK, DENS_S, DENS_PR]
Z_SIST=[Z_EGI, Z_VW, Z_RK, Z_S, Z_PR]
FIL_RVD = []
for i in range(5):
    FIL_RVD.append([EOS[i], VM_SIST[i], DENS_SIST[i], Z_SIST[i]])
print(tb.tabulate(FIL_RVD, headers=ENC_RVD, tablefmt="fancy_grid", floatfmt=".6f", stralign="center", numalign="center"))

RESUMEN DE RESULTADOS DE VOLUMEN MOLAR, DENSIDAD Y FACTOR DE COMPRESIBILIDAD DEL 
GAS PURO POR CADA ECUACION DE ESTADO
---------------------------------------------------------------------------------

╒═══════════════╤═════════════════╤════════════╤═══════════════════╕
│  ECUACIÓN DE  │  VOLUMEN MOLAR  │  DENSIDAD  │     FACTOR DE     │
│    ESTADO     │     (L/mol)     │  (kg/m^3)  │  COMPRESIBILIDAD  │
╞═══════════════╪═════════════════╪════════════╪═══════════════════╡
│   Gas ideal   │    2.769559     │  6.504645  │     1.000000      │
├───────────────┼─────────────────┼────────────┼───────────────────┤
│ Van der Waals │    2.663445     │  6.763797  │     0.961685      │
├───────────────┼─────────────────┼────────────┼───────────────────┤
│ Redlich-Kwong │    2.632088     │  6.844375  │     0.950364      │
├───────────────┼─────────────────┼────────────┼───────────────────┤
│     Soave     │    2.614312     │  6.890914  │     0.943945      │
├───────────────┼─────────────────┼────

Como se muestra en la tabla de resultados comparativa se muestra como varía la densidad y factor de compresibilidad, lo cual puede generar la pregunta de cuál ecuación de estado es mejor representa en condiciones reales ciertas compuestos en gas como por ejemplo pueden ser los hidrocarburos u otros compuestos organicos e inorgánicos en fase gaseosa. 


Ahora puedes experimentar cambiando los parámetros del componente puro ya sea en este archivo siguiendo calculando con los otros incisos, o siguiendo otros ejercicios como el siguiente.


_Determine el volumen molar, la densidad y el factor de compresibilidad para los siguientes fluidos a 11,700 kPa y 410°C para los siguientes compuestos.  
a)acetona$\;\;\;\;\;\;\;\;$b) n-heptano$\;\;\;\;\;\;\;\;$c) cloro molecular_


También experimentar entre diferentes compuestos usando este archivo y "EOS_1_compuesto_puro.ipynb" en donde se explica con más detalle las ecuaciones de estado y cambiar ciertos parámetros de referencia, utilizando el siguiente libro: Reid, R. C., Prausnitz, J. M., & Poling, B. E. (1987). The properties of gases & liquids (4 th edition). McGraw-Hill.